internal/os/disk/api.go

```diff
@@ -253,20 +254,36 @@ func (imp APIImplementor) GetDiskNumberWithID(page83ID string) (uint32, error) {
 	json.Unmarshal([]byte(outString), &disks)
 
 	for i := range disks {
-		h, err := syscall.Open(disks[i].Path, syscall.O_RDONLY, 0)
-		if err != nil {
+		diskNumber, err := imp.getDiskNumberByID(disks[i].Path, page83ID)
+		if err == ErrNotFound {
+			continue
+		} else if err != nil {
 			return 0, err
-		}
-
-		found, err := imp.DiskHasPage83ID(h, page83ID)
-		if found {
-			return imp.GetDiskNumber(h)
+		} else {
+			return diskNumber, nil
 		}
 	}
 
 	return 0, fmt.Errorf("Could not find disk with Page83 ID %s", page83ID)
 }
 
+var ErrNotFound = errors.New("not found")
+
+func (imp APIImplementor) getDiskNumberByID(path string, page83ID string) (uint32, error) {
```

Comment:
I am kind of confused by these two functions getDiskNumberAndIDByPath and getDiskNumberByID. 

Is there a way to kind of combine those?

Reply:
okay, let me combine. 

(the PR started as 2 lines, bit by bit it has ballooned, so I am getting confused too :joy:)